### DISCUSSION TITLE
New Feature Atributes "anotations" PHP 8

### DISCUSSION THREAD

**Discussion by Vagnerlg** [Respect/Validation#1428] [Ideas]
Hello

Do you plan to put the new functionality of PHP 8 "Attributes" ?
[https://www.php.net/manual/en/language.attributes.overview.php](https://www.php.net/manual/en/language.attributes.overview.php)

Example:
```php
class Email
{
    #[\Respect\Validator('email')]
    private string $email;
   ...
}
```

Thank you and congratulations for the project.

**Comment by henriquemoody** [comment]
Hi @Vagnerlg! I'm sorry for my delay!

Yes, that was on my plans for version 3.0. The way I see this rule working would be something like:

```php
use Respect\Validation\Rules\Alnum;
use Respect\Validation\Rules\Email;
use Respect\Validation\Validator;

readonly class AccountCreated
{
    #[Alnum]
    public string $username;

    #[Email]
    public string $email;

    public function __construct(string $username, string $email)
    {
        $this->username = $username;
        $this->email = $email;
    }
}

Validator::attribute()->assert(new AccountCreated('pand', 'panda@example.com'));
```


However, could you tell me a bit more about which use-cases you see for that?

**Comment by henriquemoody** [comment]
I've implemented this feature -- finally! 🎉 

It will be present in version 3.0, which I plan/hope I can get released at the beginning of next year!

https://github.com/Respect/Validation/blob/main/docs/rules/Attributes.md
